Tasks/PowerShellOnTargetMachines2/SessionHelper.ps1

```diff
@@ -0,0 +1,54 @@
+function Create-PSSessionToRemoteMachines {
+    [CmdletBinding()]
+    Param (
+        [string[]] $targetMachineNames,
+        [pscredential] $targetMachineCredential,
+        [ValidateSet("http", "https")]
+        [string] $protocol,
+        [ValidateRange(2,10)]
+        [int] $maxRetry = 3
+    )
+
+    Trace-VstsEnteringInvocation $MyInvocation
+    try {
+        $retryCount = 0;
+        $sessions = @();
+    
+        if($protocol -eq "https") {
+            $useSsl = $true
+        } else {
+            $useSsl = $false
+        }
+    
+        $remainingMachines = $targetMachineNames | ForEach-Object { $_.ToLowerInvariant() }
+    
+        while ($retryCount -lt $maxRetry) {
+            Write-Verbose "Retry attempt #$($retryCount + 1)"
+            $sessions += New-PSSession -ComputerName $remainingMachines -Credential $targetMachineCredential -UseSSL:$useSsl -ErrorAction "SilentlyContinue" -ErrorVariable sessionErrors
```

Comment:
there is an input (NewPsSessionOptionArguments) meant for this very purpose. inputs like skipcacheck skipcncheck have to be supplied there, which in turn are used as args for the New-PSSessionOption cmdlet. And instead of passing the output of that command to every invoke-command or new-possession command, i use the result to set the value of the PSSessionOption preference variable. This causes any new pssession initiated in the current session to pick these values automatically.